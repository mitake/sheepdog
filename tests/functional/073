#!/bin/bash

# Test sheepfs

. ./common

trap "_uninit; exit \$status" 0 1 2 3 15
_uninit()
{
	umount $STORE/sheepfs > /dev/null 2>&1
	rm -rf $STORE/sheepfs > /dev/null 2>&1
}

_need_to_be_root

# We can't pass PATH=$PATH:../../dog to sheepfs to expect it find dog
# because fuse main loop can't inherit $PATH passed to sheepfs
command -v dog > /dev/null 2>&1 || _notrun "Require dog but it's not installed"

if [ ! -f $SHEEPFS ]; then
    _notrun "no sheepfs binary found"
fi

for i in 0 1 2; do
    _start_sheep $i
done

_wait_for_sheep 3

_cluster_format

$DOG vdi create test 4M

mkdir $STORE/sheepfs
$SHEEPFS $STORE/sheepfs

cat $STORE/sheepfs/cluster/info | _filter_cluster_info
cat $STORE/sheepfs/vdi/list | _filter_short_date
echo test > $STORE/sheepfs/vdi/mount
echo hello > $STORE/sheepfs/volume/test
head -c 6 $STORE/sheepfs/volume/test

status=0
