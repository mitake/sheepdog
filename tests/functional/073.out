QA output created by 073
using backend plain store
Cluster status: running, auto-recovery enabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0  4.0 MB  0.0 MB  0.0 MB DATE   7c2b25     3              
hello
